src/Microsoft.Health.Fhir.Core/Models/IModelFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Hl7.Fhir.ElementModel;
+using Hl7.FhirPath;
+
+namespace Microsoft.Health.Fhir.Core.Models
+{
+    public interface IModelFactory
```

Comment:
This doesn't sound like a factory. IModelInfoProvider? Resolver?

Reply:
Going with `IModelInfoProvider`